adagrams/game.py

```diff
@@ -1,13 +1,120 @@
 from random import randint
 
 def draw_letters():
-    pass
+    # Borrowed from test_wave_01.py; used to build expanded letter pool list.
+    LETTER_POOL_FREQUENCIES = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 'I': 9,
+    'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 'Q': 1, 'R': 6,
+    'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+    }
+    MAX_WORD_LENGTH = 10  # Maximum word length permitted.
+    letter_pool = []  # will hold expanded letter pool list.
+    # iterates through dictionary, adding the appropriate number of letters to
+    # 'letter_pool' based on frequency:
+    for letter, frequency in LETTER_POOL_FREQUENCIES.items():
+        letter_pool += letter * frequency
+    hand = []  # will hold the player's drawn hand.
+
+    past_indices = []  # keeps track of what indices have been prev. selected.
+    while len(hand) < MAX_WORD_LENGTH:  # loops until 'hand' is full.
+        random_index = randint(0, len(letter_pool) - 1)  # selects random value
+        # checks if current random_index has previously been selected:
+        if random_index in past_indices:
+            continue
+        # if above check passes, adds letter to 'hand' and current index to
+        # 'past_indices':
+        hand += letter_pool[random_index]
+        past_indices.append(random_index)
+
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    formatted_word = word.upper() # formats 'word' to all uppercase letters.
+    # builds 'letter_bank_count' to keep track of letters in the bank:
+    letter_bank_count = {}
+    for letter in letter_bank:
+        if letter not in letter_bank_count:
+            letter_bank_count[letter] = 1  # adds new key:value if not present
+        else:
+            letter_bank_count[letter] += 1  # adds to value if key is present
```

Comment:
Frequency dictionaries are going to be a structure we create and use constantly in Python! They are already pretty streamlined, but we do have a way that we can pare them down even more by using the `.get()` method:

     for letter in letter_bank:
          letter_bank_count[letter] = letter_bank_count.get(letter, 0) + 1
          
As you continue creating frequency dictionaries, feel free to use this more streamlined syntax!